File: frontend/movies-app/src/app/models/movie.ts
export interface Movie {
  id?: string;
  title: string;
  director: string;
  year: number;
  genre: string;
  description: string;
  posterUrl: string;
}

export interface MoviesPage {
  items: Movie[];
  totalCount: number;
  page: number;
  pageSize: number;
}

File: backend/MoviesApi/Models/Movie.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace MoviesApi.Models;

public class Movie
{
    [BsonId]
    [BsonRepresentation(BsonType.ObjectId)]
    public string? Id { get; set; }

    public string Title { get; set; } = string.Empty;
    public string Director { get; set; } = string.Empty;
    public int Year { get; set; }
    public string Genre { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public string PosterUrl { get; set; } = string.Empty;
}